### DISCUSSION TITLE
What is Entry.SetComponent()?

### DISCUSSION THREAD

**Discussion by setanarut** [yohamta/donburi#137] [Q&A] [answered]
There are no examples in the documentation. Can you explain with an example what the usage of [Entry.SetComponent()](https://pkg.go.dev/github.com/yohamta/donburi@v1.4.2#Entry.SetComponent) is?

**Comment by yohamta0** [comment]
Hello, it's equivalent to the `donburi.Set()` method.

https://github.com/yohamta/donburi/blob/680901d36f4ad7baa7fcf059a1cfb0a5f32af14d/entry.go#L44-L47

**Reply by setanarut** [reply]
What does 'Set' mean and how is it different from adding and removing components? I really don't understand. Example please?

**Reply by yohamta0** [reply]
Here's the example. It doesn't add or remove component, but update the pointer to the data.

```go
// Component is any struct that holds some kind of data.
type PositionData struct {
  X, Y float64
}

// ComponentType represents kind of component which is used to create or query entities.
var Position = donburi.NewComponentType[PositionData]()

// Create an entity by specifying components that the entity will have.
// Component data will be initialized by default value of the struct.
entity = world.Create(Position, Velocity)

// We can use entity (it's a wrapper of int64) to get an Entry object from World
// which allows you to access the components that belong to the entity.
entry := world.Entry(entity)

entry.Set(Position, unsafe.Pointer(&PositionData{
  X: 100, Y: 200,
})

```

**Reply by setanarut** [reply]
Looks the same as component.Set().  Am I wrong?

**Reply by yohamta0** [reply, marked as answer]
Your understanding is absolutely right. It's the same as the `comopnent.Set()`.

**Reply by setanarut** [reply]
Thanks
